src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/StorageAccountBaseCmdlet.cs

```diff
@@ -76,37 +93,97 @@ public string SubscriptionId
+            return returnSkuName;
+        }
+
+        protected static Kind? ParseAccountKind(string accountKind)
+        {
+            Kind returnKind;
+            if (accountKind == null)
+            {
+                return null;
+            }
+            else if (!Enum.TryParse<Kind>(accountKind, true, out returnKind))
+            {
+                throw new ArgumentOutOfRangeException("Kind");
+            }
+            return returnKind;
+        }
+        protected static AccessTier ParseAccessTier(string accessTier)
+        {
+            AccessTier returnAccessTier;
+            if (!Enum.TryParse<AccessTier>(accessTier, true, out returnAccessTier))
             {
-                return AccountType.StandardZRS;
+                throw new ArgumentOutOfRangeException("AccessTier");
             }
-            if (AccountTypeString.StandardGRS.Equals(accountType, StringComparison.OrdinalIgnoreCase))
+            return returnAccessTier;
+        }
+
+        protected static Encryption ParseEncryption(EncryptionSupportServiceEnum? EnableService, EncryptionSupportServiceEnum? DisableService = null)
+        {
+            //DisableService and EnableService should not have overlap
+            if (DisableService != null && EnableService != null)
             {
-                return AccountType.StandardGRS;
+                if ((DisableService & EnableService) != 0)
+                    throw new ArgumentOutOfRangeException("EnableEncryptionService, DisableEncryptionService", String.Format("EnableEncryptionService and DisableEncryptionService should no have overlap Service: {0}", DisableService & EnableService));
             }
-            if (AccountTypeString.StandardRAGRS.Equals(accountType, StringComparison.OrdinalIgnoreCase))
+
+            Encryption accountEncryption = new Encryption();
+            accountEncryption.Services = new EncryptionServices();
+            if (EnableService != null && (EnableService & EncryptionSupportServiceEnum.Blob) == EncryptionSupportServiceEnum.Blob)
             {
-                return AccountType.StandardRAGRS;
+                accountEncryption.Services.Blob = new EncryptionService();
+                accountEncryption.Services.Blob.Enabled = true;
             }
-            if (AccountTypeString.PremiumLRS.Equals(accountType, StringComparison.OrdinalIgnoreCase))
+            if (DisableService != null && (DisableService & EncryptionSupportServiceEnum.Blob) == EncryptionSupportServiceEnum.Blob)
             {
-                return AccountType.PremiumLRS;
+                accountEncryption.Services.Blob = new EncryptionService();
+                accountEncryption.Services.Blob.Enabled = false;
             }
-            throw new ArgumentOutOfRangeException("accountType");
+            return accountEncryption;
+        }
+
+        public static StorageCredentials GenerateStorageCredentials(IStorageManagementClient storageClient, string resourceGroupName, string accountName)
```

Comment:
@blueww I *think* these are largely copied from commands.common.storage.  If possible, it would be better to write the version 5 adapters ans use the common functions instead.  This way other can reuse your adapter if they decide to update their storage management versions.